includes/class-ssp-frontend.php

```diff
@@ -294,13 +308,77 @@ public function get_episode_download_link( $episode_id, $referrer = '' ) {
 		return apply_filters( 'ssp_episode_download_link', esc_url( $link ), $episode_id, $file );
 	}
 
+	/**
+     * Get Album Art for Player
+     *
+     * Iteratively tries to find the correct album art based on whether the desired image is of square aspect ratio.
+     * Falls back to default album art if it can not find the correct ones.
+     *
+	 * @param $episode_id ID of the episode being loaded into the player
+     *
+	 * @return array [ $src, $width, $height ]
+     *
+     * @since 1.19.4
+	 */
+	public function get_album_art( $episode_id = false ) {
+
+	    if( !$episode_id ){
+		    $src = SSP_PLUGIN_URL . '/assets/images/no-album-art.png';
+		    $width  = 300;
+		    $height = 300;
+		    return compact( 'src', 'width', 'height' );
+        }
+
+		$series_id = false;
+
+		if ( $series = get_the_terms( $episode_id, 'series' ) ) {
+			$series_id = ( ! empty( $series ) && isset( $series[0] ) ) ? $series[0]->term_id : false;
+		}
+
+		if ( $series_id && $series_image = get_option( "ss_podcasting_data_image_{$series_id}" ) ) {
+			$series_image_attachment_id = ssp_get_image_id_from_url( $series_image );
+			list( $src, $width, $height ) = wp_get_attachment_image_src( $series_image_attachment_id, 'medium' );
+
+			if( ( $width / $height ) !== 1 ){
+				if( $series_image = get_option( "ss_podcasting_data_image" ) ) {
+					$series_image_attachment_id = ssp_get_image_id_from_url( $series_image );
+					list( $src, $width, $height ) = wp_get_attachment_image_src( $series_image_attachment_id, 'medium' );
+
+					if ( ( $width / $height ) !== 1 ) {
+						$src = SSP_PLUGIN_URL . '/assets/images/no-album-art.png';
+						$width  = 300;
+						$height = 300;
+						return compact( 'src', 'width', 'height' );
+					}
+					return compact( 'src', 'width', 'height' );
+				}
+			}
+			return compact( 'src', 'width', 'height' );
+		}elseif( $series_id && $series_image = get_option( "ss_podcasting_data_image" ) ) {
+			$series_image_attachment_id = ssp_get_image_id_from_url( $series_image );
+			list( $src, $width, $height ) = wp_get_attachment_image_src( $series_image_attachment_id, 'medium' );
+			if ( ( $width / $height ) !== 1 ) {
+				$src = SSP_PLUGIN_URL . '/assets/images/no-album-art.png';
+				$width  = 300;
+				$height = 300;
+			}
+			return compact( 'src', 'width', 'height' );
+		}else{
+			$src = SSP_PLUGIN_URL . '/assets/images/no-album-art.png';
+			$width  = 300;
+			$height = 300;
+			return compact( 'src', 'width', 'height' );
+		}
+	}
+
 	/**
 	 * Load media player for given file
 	 * @param  string  $srcFile        Source of file
 	 * @param  integer $episode_id Episode ID for audio file
+     * @param  string $player_size mini or large
 	 * @return string              Media player HTML on success, empty string on failure
 	 */
-	public function media_player ( $srcFile = '', $episode_id = 0 ) {
+	public function media_player ( $srcFile = '', $episode_id = 0, $player_size = "large" ) {
```

Comment:
Introducing a new flag to switch between the native player and the new HTML5 player. This is being done because we have replaced native audio player rendering in shortcodes to rather us this method, and we need control over the style from within the shortcode.